src/audio-manager.ts

```diff
@@ -1,261 +1,389 @@
+     *
+     * @note Is there more than one-audio file available per id, on playback, they will be selected at random.
+     * This enables easy variation of the same sounds!
      */
-    constructor(src: string, audioBuffer: AudioBuffer, audioManager: AudioManager) {
-        this._audioBuffer = audioBuffer;
-        this._audioManager = audioManager;
-        this._source = src;
-        this._gainNode.connect(_audioContext.destination);
+    async load(path: string[] | string, id: number) {
+        const paths = Array.isArray(path) ? path : [path];
+        if (!this._bufferCache.has(id)) {
+            this._bufferCache.set(id, []);
+        }
+        for (let i = 0; i < paths.length; i++) {
+            const response = await fetch(paths[i]);
+            const arrayBuffer = await response.arrayBuffer();
+            const audioBuffer = await _audioContext.decodeAudioData(arrayBuffer);
+            this._bufferCache.get(id)!.push(audioBuffer);
+        }
+
+        this.emitter.notify({id: id, state: PlayState.READY});
     }
 
     /**
-     * Asynchronously plays the loaded audio. If the audio is already playing, it stops the current playback and
-     * starts from the beginning.
-     * If the audio context is in a suspended state, it attempts to unlock the audio context before playing and will
-     * continue after the user has interacted with the website.
+     * Plays the audio file associated with the given ID.
      *
-     * @param posVec - An optional parameter representing the 3D spatial position of the audio source.
-     *                 If provided, the audio will be spatialized using a PannerNode based on the given position vector.
-     * @returns A Promise that resolves once the audio playback starts.
+     * @note Is the given ID already playing, it will restart its playback.
+     * @param id ID of the file that should be played.
+     * @param config Optional parameter that will configure how the audio is played. Is no configuration provided,
+     * the audio will play at volume 1.0, without panning and on the MASTER channel.
+     * @throws If the given ID does not have a buffer associated with it, or all players are currently occupied.
      */
-    async play(posVec?: Float32Array): Promise<void> {
-        if (this._destroy) {
-            throw 'playable-node: play() was called on destroyed node!';
+    async play(id: number, config?: PlayConfig) {
+        const buffer = this._bufferCache.get(id);
+        if (!buffer) {
+            throw `audio-manager: No identifier with number: ${id} found!`;
         }
-        if (this._isPlaying) {
-            this.stop();
-        } else if (_audioContext.state === 'suspended') {
-            await _unlockAudioContext();
+        const player = this._busyPlayers.get(id) || this._freePlayers.pop();
+        if (!player) {
+            throw 'audio-manager: Reached maximum number of simultaneously playable sounds!';
```

Comment:
I don't really want to overwrite the oldest sound. It would suck if that were the background music for example and that then just cuts out... Writing a warning to console now,i hope that is sufficient.